### ISSUE TITLE
Unexpected compile error

### ISSUE THREAD

**Issue by WeihanLi** [dotnet/roslyn#67708]
**Version Used**: 

4.5.0

**Steps to Reproduce**:

There's a sample for it: 

https://github.com/WeihanLi/SamplesInPractice/blob/f1c577cb099ceec0eab09e14725a78e8c6ae579c/RoslynSample/Program.cs

.NET 8 preview SDk is required to run the sample, and just run `dotnet run` when .NET 8 SDK had been setup

The code I'm going to compile is as follows:

``` csharp
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;
using WeihanLi.Common.Logging.Serilog;

var serviceCollection = new ServiceCollection()
        .AddLogging(builder => builder.AddSerilog())
    ;
await using var provider = serviceCollection.BuildServiceProvider();
provider.GetRequiredService<ILoggerFactory>()
    .CreateLogger("test")
    .LogInformation("Hello 1234");
```

it succeeds when I compile with `WeihanLi.Common.Logging.Serilog` NuGet package reference, while when I compile with additional NuGet package `Microsoft.Extensions.Http` with version `8.0.0-preview.2.23128.3`, it produces errors like follows:

![](https://user-images.githubusercontent.com/7604648/230728634-b8c9fa67-2e70-429f-b1a5-fd6b567cc6a5.png)

**Diagnostic Id**: CS1929

Seemed the main issue is I need to supply a runtime policy, the error message is misleading.

And how could I supply a runtime policy for the compile process?
